### PR TITLE
[docs] Add legal section with licenses, CLA, and third party software

### LINKED ISSUE #3804: [docs] Move Acknowledgements to Third party software page and create Legal section
The [Acknowledgements](https://docs.yugabyte.com/latest/architecture/concepts/acknowledgements/) page is not where open source docs usually have this page. Discuss what needs to be updated (libraries, projects, etc.), add a new legal section, and add links to doc page footers.

**Legal** section
- **YugabyteDB license** page
- **Yugabyte contributor license agreement (CLA)** page
  - [Online form](https://docs.google.com/forms/d/11hn-vBGhOZRunclC3NKmSX1cvQVrU--r0ldDLqasRIo/viewform)
  - [Google doc](https://docs.google.com/document/d/1Yu_uCcAAPhrKAtq2Bb_qj_NbHMigaLEhbpFuFwP1azk/edit)
- **Third-party licenses** page (replaces Acknowledgements page)
- **Trademarks** page (and copyrights)

### PR DESCRIPTION
**Pull request by stevebang**
Created a "Legal" section that appears under MISC in the navigation bar. Includes pages for:
- legal information (overview) with links to licenses
- Third party software

Closes #3804

### CODE REVIEW

**Comment by stevebang** [comment]
I copied the content in the existing (out-of-place and incomplete) [Acknowledgements](https://docs.yugabyte.com/latest/architecture/concepts/acknowledgements/) page to the Acknowledgements subsection of the new "Third party software" page.  I haven't removed the existing page in this PR, but will do so when given approval.

File: docs/content/latest/legal/cla.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  latest:
+    parent: legal
+    identifier: yugabyte-cla
+    weight: 40
+isTocNested: true
+showAsideToc: true
+---
+
+
+Before individual contributions to the YugabyteDB project are accepted, the contributor must complete and sign the online [Yugabyte Individual Contributor License Agreement (CLA)](https://docs.google.com/forms/d/11hn-vBGhOZRunclC3NKmSX1cvQVrU--r0ldDLqasRIo/viewform?edit_requested=true) form. A plain text copy of the agreement is shown below. You can also obtain a copy of the Individual Contributor License Agreement - V.07/26/2018 at:
```
Comment (schoudhury) [inline docs/content/latest/legal/cla.md]: remove the google doc

File: docs/content/latest/legal/cla.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+[https://docs.google.com/document/d/1Yu_uCcAAPhrKAtq2Bb_qj_NbHMigaLEhbpFuFwP1azk/edit#](https://docs.google.com/document/d/1Yu_uCcAAPhrKAtq2Bb_qj_NbHMigaLEhbpFuFwP1azk/edit#)
```
Comment (schoudhury) [inline docs/content/latest/legal/cla.md]: remove this

File: docs/content/latest/legal/cla.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+```
+Individual Contributor License Agreement – V.07/26/2018
+
+This Individual Contributor License Agreement (“Agreement”) is entered into between YugaByte, Inc., a Delaware corporation (“YugaByte,” “we” or “us” etc.) and you (as defined below). By submitting the Individual Contributor License Agreement form below, you hereby agree to the following terms for your present and future contributions submitted to YugaByte. Except for the license granted herein to YugaByte and recipients of software distributed by YugaByte, You reserve all right, title, and interest in and to Your Contributions.
```
Comment (schoudhury) [inline docs/content/latest/legal/cla.md]: change `YugaByte` to `Yugabyte` in this page

File: docs/content/latest/legal/_index.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    weight: 3000
+---
+
+## Licenses
+
+Source code in the [YugabyteDB repository](https://github.com/yugabyte/yugabyte-db/) is variously licensed under the Apache License 2.0 and the Polyform Free Trial License 1.0.0. A copy of each license can be found in the [licenses](https://github.com/yugabyte/yugabyte-db/tree/master/licenses) directory.
+
+The build produces two sets of binaries:
+
+- The entire database with all its features (including the enterprise ones) are licensed under the [Apache License 2.0](./apache-license).
```
Comment (schoudhury) [inline docs/content/latest/legal/_index.md]: don't see any reason for us to have a copy of the Apache 2.0 license text. this link to should point to https://github.com/yugabyte/yugabyte-db/blob/master/licenses/APACHE-LICENSE-2.0.txt

File: docs/content/latest/legal/_index.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+- The binaries that contain `-managed` in the artifact and help run a managed service are licensed under the [Polyform Free Trial License 1.0.0](./polyform-license).
```
Comment (schoudhury) [inline docs/content/latest/legal/_index.md]: this link should point to https://github.com/yugabyte/yugabyte-db/blob/master/licenses/POLYFORM-FREE-TRIAL-LICENSE-1.0.0.txt

File: docs/content/latest/legal/apache-license.md [hunk 6]
```diff
@@ -0,0 +1,685 @@
+---
```
Comment (schoudhury) [inline docs/content/latest/legal/apache-license.md]: remove this page

File: docs/content/latest/legal/polyform-license.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
+---
```
Comment (schoudhury) [inline docs/content/latest/legal/polyform-license.md]: remote this page

File: docs/content/latest/legal/_index.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+By default, the build options generate only the Apache License 2.0 binaries.
+
+## Individual contributor license agreement (CLA)
+
+As an open source project with a strong focus on the user community, Yugabyte welcomes contributions as GitHub pull requests. Before someone can contribute to the YugabyteDB open source project, the online form [Yugabyte Individual Contributor License Agreement](https://docs.google.com/forms/d/11hn-vBGhOZRunclC3NKmSX1cvQVrU--r0ldDLqasRIo/viewform?edit_requested=true) must be completed. To see a copy of the complete agreement, go to [Yugabyte contributor license agreement (CLA)](./cla).
```
Comment (schoudhury) [inline docs/content/latest/legal/_index.md]: link should point to https://cla-assistant.io/yugabyte/yugabyte-db

File: docs/content/latest/legal/cla.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
+---
```
Comment (schoudhury) [inline docs/content/latest/legal/cla.md]: remove this page

File: docs/content/latest/legal/_index.md (hunk 4 above)
Comment (stevebang) [reply docs/content/latest/legal/_index.md]: done

File: docs/content/latest/legal/cla.md (hunk 3 above)
Comment (stevebang) [reply docs/content/latest/legal/cla.md]: not relevant since page is removed

File: docs/content/latest/legal/_index.md (hunk 5 above)
Comment (stevebang) [reply docs/content/latest/legal/_index.md]: done

File: docs/content/latest/legal/apache-license.md (hunk 6 above)
Comment (stevebang) [reply docs/content/latest/legal/apache-license.md]: done

File: docs/content/latest/legal/polyform-license.md (hunk 7 above)
Comment (stevebang) [reply docs/content/latest/legal/polyform-license.md]: done

File: docs/content/latest/legal/cla.md (hunk 1 above)
Comment (stevebang) [reply docs/content/latest/legal/cla.md]: removed

File: docs/content/latest/legal/cla.md (hunk 2 above)
Comment (stevebang) [reply docs/content/latest/legal/cla.md]: removed

File: docs/content/latest/legal/_index.md (hunk 8 above)
Comment (stevebang) [reply docs/content/latest/legal/_index.md]: done

File: docs/content/latest/contribute/_index.md [hunk 10]
```diff
@@ -10,7 +10,7 @@ section: CONTRIBUTOR GUIDES
 menu:
   latest:
     identifier: contribute
-    weight: 2900
+    weight: 2800
```
Comment (schoudhury) [inline docs/content/latest/contribute/_index.md]: the CLA section should point to https://cla-assistant.io/yugabyte/yugabyte-db instead of the google doc.

Comment (stevebang) [reply docs/content/latest/contribute/_index.md]: Fixed

**Pull request merged**